playbooks/cluster_setup_network.yaml

```diff
@@ -63,88 +63,43 @@
   vars:
     apply_config: "{{ apply_network_config | default(false) }}"
   tasks:
-  - block:
-    - name: add ptp vlan interfaces
+    - name: Add ptp vlan interfaces - 00-ptp.netdev
       template:
-        src: "../templates/{{ item }}.j2"
-        dest: "/etc/systemd/network/{{ item }}"
+        src: "../templates/00-ptp.netdev.j2"
+        dest: "/etc/systemd/network/00-ptp.netdev"
         group: systemd-network
-      with_items:
-        - 00-ptp.netdev
-        - 01-ptp.network
-      register: ptp_interface
+      when:
+        - ptp_vlanid is defined
+      register: ptp_netdev_register
+    - name: Add ptp vlan interfaces - 01-ptp.network
+      template:
+        src: "../templates/01-ptp.network.j2"
+        dest: "/etc/systemd/network/01-ptp.network"
+        group: systemd-network
+      when:
+        - ptp_vlan_address is defined
+        - ptp_vlan_mask is defined
+      register: ptp_network_register
+    - name : "Add ptp vlan interface - 01-{{ ptp_interface }}-ptp.network"
```

Comment:
This is the part to configure the vlan for the interface ptp_interface
I suggest "Add PTP vlan on {{ ptp_interface }}"